TESTS/host_tests/timing_drift_auto.py

```diff
@@ -16,91 +16,103 @@
 """
 
 from mbed_host_tests import BaseHostTest
+import time
 
 
-class TimingDriftTest(BaseHostTest):
-    """ This test is reading single characters from stdio
-        and measures time between their occurrences.
+class TimingDriftSync(BaseHostTest):
+    """
+    This works as master-slave fashion
+    1) Device says its booted up and ready to run the test, wait for host to respond
+    2) Host sends the message to get the device current time i.e base time
     """
     __result = None
-    
-    # This is calculated later: average_drift_max * number of tick events
-    total_drift_max = None
-    
-    average_drift_max = 0.05
-    ticks = []
-    start_time = None
-    finish_time = None
-    dut_seconds_passed = None
-    total_time = None
-    total_drift = None
-    average_drift = None
-    
+    mega = 1000000.0
+    max_measurement_time = 180
+
+    # this value is obtained for measurements when there is 0 transport delay and we want accurancy of 5%
+    time_measurement_multiplier = 80
+
     def _callback_result(self, key, value, timestamp):
         # We should not see result data in this test
         self.__result = False
 
     def _callback_end(self, key, value, timestamp):
         """ {{end;%s}}} """
         self.log("Received end event, timestamp: %f" % timestamp)
-        self.notify_complete(result=self.result(print_stats=True))
+        self.notify_complete(result=self.result())
 
+    def _callback_timing_drift_check_start(self, key, value, timestamp):
+        self.round_trip_base_start = timestamp
+        self.send_kv("base_time", 0)
 
-    def _callback_tick(self, key, value, timestamp):
-        """ {{tick;%d}}} """
-        self.log("tick! %f" % timestamp)
-        self.ticks.append((key, value, timestamp))
+    def _callback_base_time(self, key, value, timestamp):
+        self.round_trip_base_end = timestamp
+        self.device_time_base = float(value)
+        self.round_trip_base_time = self.round_trip_base_end - self.round_trip_base_start
 
+        self.log("Device base time {}".format(value))
+        measurement_stretch = (self.round_trip_base_time * self.time_measurement_multiplier) + 5
+
+        self.log("Time required to measure drift accurately  : {} ".format(measurement_stretch))
+        if measurement_stretch > self.max_measurement_time:
+            self.log("Measurement time is high to accurately qualify device timer")
```

Comment:
The log message is a bit vague here. Could it be: "Time required to determine device timer is too high due to transport delay, skipping"